src/request.js

```diff
@@ -62,7 +62,7 @@ export default class Request {
 				parsedURL = parse_url(input.href);
 			} else {
 				// coerce input to a string before attempting to parse
-				parsedURL = parse_url(`${input}`);
+				parsedURL = parse_url(encodeURI(`${input}`));
```

Comment:
There's a possibility that the url is already encoded, in which case it would be encoded twice. Would that end up being an issue?

Adding a test case with an already-encoded url might be a good idea.

Reply:
Yeah, you're right. The `URL` constructor handles this so we must too 👍

Reply:
Fixed ✅